sprokit/src/bindings/python/modules/registration.cxx

```diff
@@ -105,13 +115,24 @@ register_factories(kwiver::vital::plugin_loader& vpm)
     PySys_SetArgvEx(0, NULL, 0);
   }
 
-  _load_python_library_symbols();
-
-  sprokit::python::python_gil const gil;
+  if (!PyEval_ThreadsInitialized())
+  {
+    // When this is destructed, PyEval_SaveThread() will be called to release
+    // the GIL
+    scoped_save_thread save_thread;
+    (void)save_thread;
+    {
+      // Let pybind11 initialize threads and set up its internal data structures
+      pybind11::gil_scoped_acquire acquire;
+      (void)acquire;
+    }
+  }
 
-  (void)gil;
+  _load_python_library_symbols();
 
+  SPROKIT_PYTHON_GIL_SCOPED_ACQUIRE_BEGIN
```

Comment:
per Linus' comment on https://github.com/Kitware/kwiver/pull/459 (which I abandoned) we might want to swap these macros out so they don't have an explicit start/end, I think it's fine either way. It looks like everywhere they're used you might be able to just use the gil_scoped_acquire in non-macro form to simplify the code.

Reply:
I decided to go with the macros because I don't want to duplicate the `(void)acquire;` warning suppression everywhere, but I am also ok with doing it either way.

Reply:
You can still wrap the warning suppression in a macro. I just think that it is easy for a future maintainer to get the ..._END macro in the wrong scope, either by inserting more complicated logic into the scoped block or not understanding the semantics of the ..._END macro. Most syntax aware editors will not show the implied scope created by the ..._BEGIN and ..._END macros. A macro something like what follows could be used to acquire the GIL in a { ... } scope.

    #define SPROKIT_PYTHON_GIL_SCOPED_ACQUIRE \
       pybind11::gil_scoped_acquire acquire;  (void)acquire

Reply:
Is the warning suppression strictly necessary? Getting rid of it would eliminate the need for macros entirely and simplify this a lot.